python/rascaline/rascaline/utils/clebsch_gordan/_cg_cache.py

```diff
@@ -445,10 +639,28 @@ def sparse_combine(
     n_p = arr_1.shape[2]  # number of properties in arr_1
     n_q = arr_2.shape[2]  # number of properties in arr_2
 
-    if return_empty_array:  # used when only computing metadata
-        return _dispatch.zeros_like((n_i, 2 * lambda_ + 1, n_p * n_q), like=arr_1)
+    # The isinstance checks and cg_backend checks makes the logic a bit redundant
+    # but the redundancy by the isinstance check is required for TorchScript. Logic
+    # can be made more straightforward once MOPS support TorchScript
```

Comment:
so the plan for mops is to have a `mops-torch` thing as well. For now you can just error out if `mop == true and TorchScript == True`